src/slack_model.jl

```diff
@@ -5,32 +5,47 @@ export SlackModel, SlackNLSModel,
        hess_coord, hess, hprod, hprod!
 
 
-"""A model whose only inequality constraints are bounds.
+@doc raw"""A model whose only inequality constraints are bounds.
 
 Given a model, this type represents a second model in which slack variables are
 introduced so as to convert linear and nonlinear inequality constraints to
 equality constraints and bounds. More precisely, if the original model has the
 form
 
-\\\\[ \\min f(x)  \\mbox{ s. t. }  c_L \\leq c(x) \\leq c_U \\mbox{ and }
-\\ell \\leq x \\leq u, \\\\]
+```math
+\begin{align}
+&\min_x      & &f(x)\\
+&\text{s.t.} & &\begin{array}{ccccc}
+                  c_L &≤& c(x) &≤& c_U,\\
+                   l  &≤&   x  &≤&  u,
+                \end{array}
+\end{align}
+```
 
 the new model appears to the user as
 
-\\\\[ \\min f(X)  \\mbox{ s. t. }  g(X) = 0 \\mbox{ and } L \\leq X \\leq U. \\\\]
+```math
```

Comment:
I believe `\begin{align}` is missing here, and possibly the `array`, so that it's similar to the expression above.

Reply:
I can't see how to align the equation `g(x) = 0` with chained inequality `L ≤ X ≤ U` in `array`. But with/without `array` does create some discrepancy. The `array` environment is creating extra spaces between columns:
<img src="https://user-images.githubusercontent.com/22480702/55294992-91b91780-53c5-11e9-9a31-2d4448ca18e7.png" alt="array" width="40%">
 and I cannot find a way to remove that yet. (I tried `\setlength\arraycolsep{1pt}` and `\begin{array}[t]{c@{\hspace{1pt}}c...}`, both work with Latex but not with Documenter.jl). 

A workaround is to use `alignedat`, which results in something like this:
<img src="https://user-images.githubusercontent.com/22480702/55295084-929e7900-53c6-11e9-83d1-f8f29487d3b2.png" alt="alignedat" width="30%" >
I guess this is the best we can have.

Reply:
I think we can use the same format as in the docs (https://github.com/JuliaSmoothOptimizers/NLPModels.jl/blob/master/docs/src/index.md), i.e., simple `align`.

Reply:
I'll leave it with `alignedat` for now. I think it's slightly better than just `align`. Let me know if you prefer `align`.

Reply:
Sorry for the delay. What I meant was changing to something simpler overall, like this:
```
\begin{align*}                                                                              
       \min_x \quad & f(x) \\                                                               
\mathrm{s.t.} \quad & c_L ≤ c(x) ≤ c_U,\\                                                   
                    &   ℓ ≤   x  ≤ u,                                                      
\end{align*}                                                                                
```
which uses only a simple align, and is also easy to read on the help.

Reply:
Less is more. Got it. Do I need to care about the display maths in markdown files?

Reply:
Sorry, what do you mean?

Reply:
Forget it. I was thinking about updating the maths in docs/*.md like realigning and using Unicode. But since markdown files will not appear in REPL as source code, I guess we can just leave them alone.